terminal.go

```diff
@@ -118,20 +162,16 @@ func readOrCreateHistory(f string) ([]string, error) {
 }
 
 func saveHistory(f string, h []string) {
```

Comment:
I would change to


```suggestion
 func saveHistory(f string, h []string) error {
```

To do not log here but wrap error, the log in Close

Reply:
I typically prefer to log the closest the error occurred unless I expect the caller to want to handle it themselves. in this case that function is private exclusively used by `Close()` and there is no error returning by definition because it's ran "at the end" as a `defer`